modules/test/playwright/tests/frontend-data-set-web/main/tests/advanced/actions.spec.ts

```diff
@@ -71,6 +71,94 @@ test('Behavior of item actions', async ({fdsSamplePage, page}) => {
 		await page.keyboard.press('Escape');
 	});
 
+	await test.step('Check that the Item Actions dropdown displays icons for Table, List, and Cards views', async () => {
+		await test.step('Check Table view actions dropdown items has icons', async () => {
+			const dropdownId =
+				await itemActionButton.getAttribute('aria-controls');
+
+			await itemActionButton.click();
+
+			const dropdownMenu = page.locator(`#${dropdownId}`);
+
+			await dropdownMenu.filter({has: page.getByRole('menu')}).waitFor();
+
+			const menuItems = dropdownMenu.getByRole('menuitem');
+
+			for (const menuItem of await menuItems.all()) {
+				await expect(menuItem.locator('.lexicon-icon')).toBeVisible();
+			}
+
+			await page.keyboard.press('Escape');
+		});
+
+		await test.step('Check List view actions dropdown items has icons', async () => {
+			await fdsSamplePage.changeVisualizationMode({
+				page,
+				visualizationMode: EFDSVisualizationMode.LIST,
+			});
+
+			const listItemActionButton = fdsSamplePage.list.items
+				.first()
+				.getByRole('button', {
+					exact: true,
+					name: 'Actions',
+				});
+
+			await listItemActionButton.click();
+
+			const dropdownMenu = page.locator(
+				`#${await listItemActionButton.getAttribute('aria-controls')}`
+			);
+
+			await dropdownMenu.filter({has: page.getByRole('menu')}).waitFor();
+
+			const menuItems = dropdownMenu.getByRole('menuitem');
+
+			for (const menuItem of await menuItems.all()) {
+				await expect(menuItem.locator('.lexicon-icon')).toBeVisible();
+			}
```

Comment:
looks like there's a bunch of common code once we know what's the dropdownMenu locator:

```
			await dropdownMenu.filter({has: page.getByRole('menu')}).waitFor();

			const menuItems = dropdownMenu.getByRole('menuitem');

			for (const menuItem of await menuItems.all()) {
				await expect(menuItem.locator('.lexicon-icon')).toBeVisible();
			}
```

As we want to stress that actions work the same across the visualizations, it sounds natural to me if these are abstracted out into a funcion called by all test steps.

We could take it a bit further if we abstract out the viz mode change and identification of the actions button but that would complicate code much more.

What do you think?

Reply:
Ya, I think some sort of `checkMenuIconsAreVisible()" function makes sense since they're checked the same way.

As for the viz mode + action button, I think it makes sense for the action button to be a defined locator in the `FDSSamplePage` like `fdsSamplePage.cards.getActionButton(index)`. I don't find any value in grouping the viz mode change and getting the action button since they seem specific to the test.

I'll apply these changes though! Thanks for the suggestion!